scene/gui/graph_edit.cpp

```diff
@@ -1666,14 +1666,81 @@ HashMap<int, Vector<StringName>> GraphEdit::layering(const Set<StringName> &v, c
 		}
 		selected = false;
 	}
+	
+	return l;
+}
+
+void GraphEdit::crossing_minimisation(HashMap<int, Vector<StringName>>& l, const HashMap<StringName, Set<StringName>>& u) {
+	if (l.size() == 1) {
+		return;
+	}
+
 	//Print layers here
 	//Check if vectors of hashmap are being written
-	for (int i = 1; l.has(i); i++) {
+	for (int i = 0; l.has(i); i++) {
 		for (int j = 0; j < l[i].size(); j++) {
 			StringName temp = l[i][j];
 		}
 	}
-	return l;
+
+	for (int i = 1; i < l.size(); i++) {
+		Vector<StringName> upper_layer = l[i - 1];
+		Vector<StringName> lower_layer = l[i];
+		HashMap<StringName, HashMap<StringName, int>> c;
+
+		for (int j = 0; j < lower_layer.size(); j++) {
+			StringName p = lower_layer[j];
+			HashMap<StringName, int> cross;
+
+			for (int k = 0; k < lower_layer.size(); k++) {
+				int crossings = 0;
+				StringName q = lower_layer[k];
+
+				if (j == k) {
+					cross.set(q, crossings);
+				}
+				else if (j < k) {
```

Comment:
I prefer reversed conditions and I believe this doesn't follow the coding standard and the cicd will block the merge.

```
if (!var) {
    continue;
}

Reply:
There's a new line after } else. I'm referring to this.

Reply:
Will fix it right away. Forgot to ran document formatter before pushing! My bad...